main/lsp/requests/completion.cc

```diff
@@ -261,12 +262,26 @@ unique_ptr<CompletionItem> LSPLoop::getCompletionItemForSymbol(const core::Globa
         if (what.exists()) {
             item->detail = methodDetail(gs, what, receiverType, nullptr, constraint);
         }
+
+        u4 queryStart = queryLoc.beginPos();
+        u4 prefixSize = prefix.size();
+        auto replacementLoc = core::Loc{queryLoc.file(), queryStart - prefixSize, queryStart};
+        auto replacementRange = Range::fromLoc(gs, replacementLoc);
+
+        string replacementText;
         if (config.clientCompletionItemSnippetSupport) {
             item->insertTextFormat = InsertTextFormat::Snippet;
-            item->insertText = methodSnippet(gs, what);
+            replacementText = methodSnippet(gs, what);
         } else {
             item->insertTextFormat = InsertTextFormat::PlainText;
-            item->insertText = string(what.data(gs)->name.data(gs)->shortName(gs));
+            replacementText = string(what.data(gs)->name.data(gs)->shortName(gs));
+        }
+
+        if (replacementRange != nullptr) {
+            item->textEdit = make_unique<TextEdit>(std::move(replacementRange), replacementText);
+        } else {
+            // TODO(jez) Why is replacementRange nullptr? instrument this and investigate when it fails
```

Comment:
This does seem surprising.

Reply:
To be clear, I haven't seen this be `nullptr`, I just cargo culted from other places in the codebase where it was. I think that we could make this be an ENFORCE, but it wouldn't fire in release builds so we'd never know about it. I'm going to follow up with some observability here that will alert us when this triggers.